cmd/kubeadm/app/preflight/checks_test.go

```diff
@@ -692,3 +693,38 @@ func TestSetHasItemOrAll(t *testing.T) {
 		}
 	}
 }
+
+type imgs struct{}
+
+func (i *imgs) Pull(image string) error {
+	if image == "d" {
+		return errors.New("pull error")
+	}
+	return nil
+}
+func (i *imgs) Exists(image string) (int, error) {
+	switch image {
+	case "a":
+		return 0, nil
```

Comment:
for readability, could you add a comment on each case here, like
`// image a not present locally, should be pulled`
`// image b cached locally, should not be pulled`
`// image c failed to inspect`
etc.